### ISSUE TITLE
Preserve parameters in each class and not all of the test parameters

### ISSUE THREAD

**Issue by ausias-armesto** [cbeust/testng#2008]
### TestNG Version
> 7.0.1-beta and previous

### Expected behavior
Given a fragment of suite.xml like the following:

```
...
<test thread-count="1" verbose="0" name="testname" parallel="instances">
    <classes>
      <class name="com.dummy.DummyTestA">
        <parameter name="parameterName1" value="Value 1"/>
        <parameter name="parameterName2" value="Value 2"/>
        <methods>
		...
        </methods>
      </class>
    <classes>
      <class name="com.dummy.DummyTestB">
        <parameter name="parameterName2" value="OverrideValue 2"/>
        <parameter name="parameterName3" value="Value 3"/>
        <methods>
		...
        </methods>
      </class>	  
    </classes>
  </test>
...
```
If both classes execution fail, then the FailedReport should keep the parameters for each class and not mix them so should generate the following testng-failed.ml

```
...
<test thread-count="1" verbose="0" name="testname(failed)" parallel="instances">
    <classes>
      <class name="com.dummy.DummyTestA">
        <parameter name="parameterName1" value="Value 1"/>
        <parameter name="parameterName2" value="Value 2"/>
        <methods>
		...
        </methods>
      </class>
    <classes>
      <class name="com.dummy.DummyTestB">
        <parameter name="parameterName2" value="OverrideValue 2"/>
        <parameter name="parameterName3" value="Value 3"/>
        <methods>
		...
        </methods>
      </class>	  
    </classes>
  </test>
...
```

### Actual behavior

But instead the FailedReport produces the following testng-failed.ml

```
...
<test thread-count="1" verbose="0" name="testname(failed)" parallel="instances">
    <classes>
      <class name="com.dummy.DummyTestA">
        <parameter name="parameterName1" value="Value 1"/>
        <parameter name="parameterName2" value="OverrideValue 2"/>
		<parameter name="parameterName3" value="Value 3"/>
        <methods>
		...
        </methods>
      </class>
    <classes>
      <class name="com.dummy.DummyTestB">
		<parameter name="parameterName1" value="Value 1"/>
        <parameter name="parameterName2" value="OverrideValue 2"/>
        <parameter name="parameterName3" value="Value 3"/>
        <methods>
		...
        </methods>
      </class>	  
    </classes>
  </test>
...
```
Mixing the parameters for both classes and their values.

### Is the issue reproductible on runner?

- [ ] Shell
- [X] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [ ] IntelliJ
- [ ] NetBeans

### Code

Looking at the source code I have seen that it is not implemented at the moment to preserve the parameters. On method org.testng.reporters.FailedReporter.createXmlClasses() we can see this:

```
...
    // Ideally, we should preserve each parameter in each class but putting them
    // all in the same bag for now
    Map<String, String> parameters = Maps.newHashMap();
    for (XmlClass c : srcXmlTest.getClasses()) {
      parameters.putAll(c.getLocalParameters());
    }
....
      xmlClass.setParameters(parameters);

```

**Comment by krmahadevan** [comment]
Since you have gone through the process of researching the root cause why not raise a pull request to fix the problem?

**Comment by krmahadevan** [comment]
Fixed via #2010
